src/Microsoft.AspNetCore.OData/Batch/ODataBatchHttpRequestMessageExtensions.cs

```diff
@@ -188,12 +191,21 @@ internal static Task CreateODataBatchResponseAsync(this HttpRequest request, IEn
             writerSettings.MessageQuotas = messageQuotas;
 
             HttpResponse response = request.HttpContext.Response;
-            response.StatusCode = (int)HttpStatusCode.OK;
 
-            // Need to get the stream from ODataBatchContent.
-            // maybe use the shared class to define bahaviour and put
-            // HttpContent and public stream getter in platform-specific?
-            ODataBatchContent batchContent = new ODataBatchContent(responses, requestContainer);
+            StringValues acceptHeader = request.Headers["Accept"];
+            string contentType = null;
+            if (StringValues.IsNullOrEmpty(acceptHeader)
+                || acceptHeader.Any( h => h.Equals(ODataBatchHttpRequestExtensions.BatchMediaTypeMime, StringComparison.OrdinalIgnoreCase)))
+            {
+                contentType = String.Format(CultureInfo.InvariantCulture, "multipart/mixed;boundary=batchresponse_{0}", Guid.NewGuid());
+            }
+            else if (acceptHeader.Any( h => h.Equals(ODataBatchHttpRequestExtensions.BatchMediaTypeJson, StringComparison.OrdinalIgnoreCase)))
+            {
+                contentType = ODataBatchHttpRequestExtensions.BatchMediaTypeJson;
+            }
```

Comment:
> } [](start = 11, length = 2)

what about "else" ? 

Maybe set "contextType" as default

Reply:
For 'else', it is an no-op for batch processing.

---
In reply to: [202868150](https://github.com/OData/WebApi/pull/1536#discussion_r202868150) [](ancestors = 202868150)